packages/element-web-opendesk-module/src/global.d.ts

```diff
@@ -16,6 +16,47 @@
 
 import 'styled-components';
 
+type Container = 'right' | 'top';
+
+type Room = {
+  roomId: string;
+};
+
+type Widget = {
+  avatar_url?: string;
+  id: string;
+  name?: string;
+  type: string;
+};
+
+declare global {
+  interface Window {
+    mxWidgetLayoutStore: {
+      isInContainer: (
+        room: Room,
+        widget: Widget,
+        container: Container,
+      ) => boolean;
+      moveToContainer: (
+        room: Room,
+        widget: Widget,
+        container: Container,
+      ) => void;
+    };
+    mxWidgetStore: {
+      getApps: (roomId: string) => Array<Widget>;
+      matrixClient?: {
+        getRoom: (roomId: string | undefined) => Room | null;
+        mxcUrlToHttp: (
+          mxcUrl: string,
+          width?: number,
+          height?: number,
+        ) => string | null;
+      };
+    };
+  }
+}
+
```

Comment:
The module API allows adding extra icons/buttons to the room header. It basically doesn't care or know that we are placing widgets here.

I thought about pulling solely the types in from the React SDK, but didn't find a good way. Since, we (Dominik and I) don't want to to have the React SDK as a dependency here.

Reply:
We will handle this separately according to other priorities.